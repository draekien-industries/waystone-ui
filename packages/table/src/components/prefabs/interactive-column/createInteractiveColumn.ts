import type { ColumnDef } from '@tanstack/react-table';
import { InteractiveColumnCell } from './interactiveColumnCell';
import { InteractiveColumnHeader } from './interactiveColumnHeader';

export const createInteractiveColumn = <D = unknown, V = unknown>(
  column?: Omit<ColumnDef<D, V>, 'header' | 'cell'>
): ColumnDef<D, V> => {
  const base = {
    id: 'table-interactions',
    header: InteractiveColumnHeader,
    cell: InteractiveColumnCell,
  };

  return column
    ? {
        ...base,
        ...column,
      }
    : base;
};
