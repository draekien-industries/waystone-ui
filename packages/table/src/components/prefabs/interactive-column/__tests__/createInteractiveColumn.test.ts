import { createInteractiveColumn } from '../createInteractiveColumn';

describe('createInteractiveColumn', () => {
  it('should match snapshot', () => {
    expect(createInteractiveColumn()).toMatchSnapshot();
  });

  it('should use column overrides', () => {
    const column = createInteractiveColumn({
      id: 'custom-id',
      enableGrouping: true,
    });

    expect(column).toMatchSnapshot();
    expect(column.id).toBe('custom-id');
    expect(column.enableGrouping).toBeTruthy();
  });
});
