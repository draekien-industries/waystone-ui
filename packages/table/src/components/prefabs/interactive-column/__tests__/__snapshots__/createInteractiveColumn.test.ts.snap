// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createInteractiveColumn should match snapshot 1`] = `
{
  "cell": [Function],
  "header": [Function],
  "id": "table-interactions",
}
`;

exports[`createInteractiveColumn should use column overrides 1`] = `
{
  "cell": [Function],
  "enableGrouping": true,
  "header": [Function],
  "id": "custom-id",
}
`;
